src/main/java/com/powsybl/openloadflow/network/impl/LfDanglingLineBranch.java

```diff
@@ -75,8 +75,14 @@ public BranchResult createBranchResult(double preContingencyP1, double branchInC
     }
 
     @Override
-    public void updateState(boolean phaseShifterRegulationOn, boolean isTransformerVoltageControlOn) {
-        danglingLine.getTerminal().setP(p.eval() * PerUnit.SB);
-        danglingLine.getTerminal().setQ(q.eval() * PerUnit.SB);
+    public void updateState(boolean phaseShifterRegulationOn, boolean isTransformerVoltageControlOn, boolean dc) {
+        if (this.isZeroImpedanceBranch(dc)) {
+            LfBus boundaryBus = getBus1() instanceof LfDanglingLineBus ? getBus1() : getBus2();
```

Comment:
I think we don't need to check this, as by construction, LfDanglingLineBus is always on side 2